### PR TITLE
Add square mask for workspace image upload

### PR DESCRIPTION
**Pull request by PrashantMangukiya**
@parasharrajat @arosiclair PR is ready for review

### Details
Changed shape of upload photo visible area for workspace image mask to square, so now **when we upload workspace image it will show square mask with rounded corner**, while for **profile image it will show circle mask.**

### Fixed Issues
$ https://github.com/Expensify/App/issues/15845
PROPOSAL: https://github.com/Expensify/App/issues/15845#issuecomment-1464329734

### Tests
1. Open the app
2. Open settings
3. Open Workspaces
4. Open any workspace
5. Click workspace profile
6. Again click on profile and click upload photo
7. Select any image and click okay, observe profile mask visible area shape, **it should be square with rounded corner**. (Previously it was circle)
8. Go to Settings > Profile > Upload Profile Image
9. Verify that **profile image mask should be circle**

- [x] Verify that no errors appear in the JS console

### Offline tests
1. Open the app
2. Open settings
3. Open Workspaces
4. Open any workspace
5. Click workspace profile
6. Again click on profile and click upload photo
7. Select any image and click okay, observe profile visible area shape, **it should be square with rounded corner**. (Previously it was circle)
8. Go to Settings > Profile > Upload Profile Image
9. Verify that **profile image mask should be circle**

### QA Steps
1. Open the app
2. Open settings
3. Open Workspaces
4. Open any workspace
5. Click workspace profile
6. Again click on profile and click upload photo
7. Select any image and click okay, observe profile visible area shape, **it should be square with rounded corner**. (Previously it was circle)
8. Go to Settings > Profile > Upload Profile Image
9. Verify that **profile image mask should be circle**

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

#### Workspace Image Upload (Square Mask)

Web-WorkspaceImage

#### Profile Image Upload (Circle Mask)

Web-ProfileImage

</details>

<details>
<summary>Mobile Web - Chrome</summary>

#### Workspace Image Upload (Square Mask)

MobileWeb-Chrome-Workspace

#### Profile Image Upload (Circle Mask)

MobileWeb-Chrome-Profile

</details>

<details>
<summary>Mobile Web - Safari</summary>

#### Workspace Image Upload (Square Mask)

MobileWeb-Safari-Workspace

#### Profile Image Upload (Circle Mask)

MobileWeb-Safari-Profile

</details>

<details>
<summary>Desktop</summary>

#### Workspace Image Upload (Square Mask)

Desktop-Workspace

#### Profile Image Upload (Circle Mask)

Desktop-Profile

</details>

<details>
<summary>iOS</summary>

#### Workspace Image Upload (Square Mask)

iOS-Workspace

#### Profile Image Upload (Circle Mask)

iOS-Profile

</details>

<details>
<summary>Android</summary>

#### Workspace Image Upload (Square Mask)

Android-Workspace

#### Profile Image Upload (Circle Mask)

Android-Profile

</details>

### CODE REVIEW

**Review by parasharrajat — CHANGES_REQUESTED** [review]

File: src/styles/styles.js [hunk 1]
```diff
@@ -2821,6 +2821,12 @@ const styles = {
         cursor: 'move',
     },
 
+    imageCropViewBorder: {
+        borderRadius: variables.componentBorderRadiusCard,
+        borderWidth: 2,
+        borderColor: '#ffffff',
```
[the remark this answers is not in the document]
Comment (PrashantMangukiya) [reply src/styles/styles.js]: Done

File: src/styles/styles.js [hunk 2]
```diff
@@ -2821,6 +2821,12 @@ const styles = {
         cursor: 'move',
     },
 
+    imageCropViewBorder: {
+        borderRadius: variables.componentBorderRadiusCard,
+        borderWidth: 2,
+        borderColor: themeColors.text,
```
[the remark this answers is not in the document]
Comment (PrashantMangukiya) [reply src/styles/styles.js:2827]: I did not get exactly which variable we have to change. Can you please let me in details.

[the remark this answers is not in the document]
Comment (PrashantMangukiya) [reply src/styles/styles.js:2827]: Created variable named cropViewBorderRadius: 12 and used for borderRadius

File: src/styles/styles.js [hunk 3]
```diff
@@ -2821,6 +2821,12 @@ const styles = {
         cursor: 'move',
     },
 
+    imageCropViewBorder: {
+        borderRadius: variables.cropViewBorderRadius,
+        borderWidth: 2,
+        borderColor: themeColors.text,
```
[the remark this answers is not in the document]
Comment (PrashantMangukiya) [reply src/styles/styles.js]: Thanks, corrected

**Comment by arosiclair** [comment]
Yeah the code would be noticeably cleaner if we passed SVG masks directly as a prop. Shawn made a good point about not necessarily needing SVG though.

@shawnborton do you have bandwidth to whip up that rounded square mask for us? If no, no worries we can move forward as is it's not a critical change.

**Comment by PrashantMangukiya** [comment]
I am ready to update code as soon as receive rounded square svg.

**Comment by PrashantMangukiya** [comment]
Thanks let me try

**Comment by PrashantMangukiya** [comment]
@shawnborton It looks as under

#### Profile Image
Profile

Workspace Image:
Workspace

Should we change corner radius within css? or  Correct mask radius?. At present within css corner radius is 12

**Comment by PrashantMangukiya** [comment]
Waiting for reply  ⬆️  @shawnborton  @parasharrajat

**Comment by shawnborton** [comment]
Yeah, maybe we should remove the border radius from the outer wrapper. Also, where did the border go?

**Comment by PrashantMangukiya** [comment]
white stroke was there in old svg. But new svg does not have any stroke. Attached is the old circle mask svg.
[OldMask.zip](https://github.com/Expensify/App/files/11005429/OldMask.zip) 

Here is old svg code:
```
<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 25.4.1, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" fill="none" 
    viewBox="0 0 407 407" style="enable-background:new 0 0 407 407;" xml:space="preserve">
<path fill-rule="evenodd" clip-rule="evenodd" d="M203.5 0H0V203.5V407H203.5H407V203.5V0H203.5ZM203.5 0C315.89 0 407 91.11 407 203.5C407 315.89 315.89 407 203.5 407C91.11 407 0 315.89 0 203.5C0 91.11 91.11 0 203.5 0Z" fill="black" fill-opacity="0.75"/>
<circle cx="203.5" cy="203.5" r="202.5" stroke="white" stroke-width="2" fill="white" fill-opacity="0"/>
</svg>
```
Can you please send me svg with white stroke similar to old svg.

**Comment by shawnborton** [comment]
Ah, my bad. One moment.

**Comment by shawnborton** [comment]
Let's try these: [masks.zip](https://github.com/Expensify/App/files/11005781/masks.zip)

**Comment by PrashantMangukiya** [comment]
Profile

Workspace

**Comment by shawnborton** [comment]
Nice, looks good to me.

**Comment by PrashantMangukiya** [comment]
Thanks.I am going to push the changes.

**Comment by PrashantMangukiya** [comment]
@parasharrajat @arosiclair I pushed code that uses svg mask as props. Also updated all screenshots. You can now continue pr review.

**Review by parasharrajat — CHANGES_REQUESTED** [review]

File: src/components/AvatarCropModal/AvatarCropModal.js [hunk 4]
```diff
@@ -50,6 +50,9 @@ const propTypes = {
     /** Modal visibility */
     isVisible: PropTypes.bool.isRequired,
 
+    /** Image crop vector mask */
+    maskImage: PropTypes.func,
```
[the remark this answers is not in the document]
Comment (PrashantMangukiya) [reply src/components/AvatarCropModal/AvatarCropModal.js:54]: We are using proptype as func for fallback icon as shown below.
https://github.com/Expensify/App/blob/a00e18dcd0ea4a77902b51cfc83366fbf73c5c56/src/components/AvatarWithImagePicker.js#L57-L58

I am not sure which prop type is feasible other than func. Prop type "object" is forbidden by eslint so we can not use it. As per your opinion which prop type should we use?

**Comment by PrashantMangukiya** [comment]
@parasharrajat I pushed changes as requested. Not sure about proper prop type for `maskImage: PropTypes.func` so it is unchanged at present. Thanks.

File: src/components/Icon/Expensicons.js [hunk 5]
```diff
@@ -163,7 +164,8 @@ export {
     Hashtag,
     History,
     Hourglass,
-    ImageCropMask,
+    CircleCropMask,
+    ImageCropSquareMask,
```
[the remark this answers is not in the document]
Comment (PrashantMangukiya) [reply src/components/Icon/Expensicons.js]: Ok changed `CircleCropMask` -> `ImageCropCircleMask` and related references.

**Review by parasharrajat — APPROVED** [review]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

cc: @arosiclair 

:ribbon: :eyes: :ribbon:  C+ reviewed

**Review by arosiclair — APPROVED** [review]
Nice work thanks for the updates!

**Pull request merged**